navigator/README.md

```diff
@@ -0,0 +1,350 @@
+# Navigator
+
+This library sits on top of AndroidX Navigation and solves 2 problems that we ran into after
+starting to adopt it.
+
+1. We believe that navigation should be triggered from the business logic. This way it is easily
+   testable where it is easily testable in isolation without having to actually navigate inside a
+   running app. However you usually want to keep Android specific components out of these layers
+   and passing `NavController` there would risk leaking the Activity.
+
+2. The XML based declaration of destinations did not scale well with our heavily modularized code
+   base. Each screen is usually it's own feature module and those should stay independent of each
+   other. That effectively meant that each XML file had a single destination and you need to define
+   all ids somewhere else so that feature modules can navigate to each other. Since the code
+   generated by safe-args also lives in the feature module it's not usable by other features
+   either. That meant that in the end we didn't get the nice type safety advantages of safe-args
+   and the XML was just boilerplate that didn't form a real graph in the end.
+
+## Destinations
+
+To replace the XML and get type safety for navigation the library has it's own concept of
+destinations. A destination consists of 2 things the declaration of the destination itself
+which determines what is shown when the destination is navigated to and a way route to reach the
+destination.
+
+### NavRoute
+
+The route part is represented by the `NavRoute` interface. Each destination will get it's own
+implementation of this interface and screens can use it to navigate to the destination.
+
+The most minimal implementation of `NavRoute` would be for a screen that doesn't require any
+arguments can be a simple Kotlin object:
+```kotlin
+@Parcelize
+object HomeScreenRoute : NavRoute, Parcelable
+```
+
+The more common case when a destination needs arguments passed to it would look like this
+```kotlin
+@Parcelize
+data class DetailScreenRoute(
+    val id: String,
+) : NavRoute, Parcelable
+```
+
+Both of these use `Parcelable` and `Parcelize` because internally PLACEHOLDER will pass the route
```

Comment:
Yes. I first wrote placeholder in case I manage to come up with a more unique name for the library and missed that

```suggestion
Both of these use `Parcelable` and `Parcelize` because internally the library will pass the route
```